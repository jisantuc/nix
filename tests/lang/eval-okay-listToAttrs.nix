# this test shows how to use listToAttrs and that evaluation is still lazy (throw isn't called)
let 
  asi = attr: value : { inherit attr value; };
  list = [ ( asi "a" "A" ) ( asi "b" "B" ) ];
  a = builtins.listToAttrs list;
  b = builtins.listToAttrs ( list ++ list );
  r = builtins.listToAttrs [ (asi "result" [ a b ]) ( asi "throw" (throw "this should not be thrown")) ];
in r.result
