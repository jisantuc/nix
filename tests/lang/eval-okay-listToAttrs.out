List([Attrs([Bind("a",Str("A",[]),NoPos),Bind("b",Str("B",[]),NoPos)]),Attrs([Bind("a",Str("A",[]),NoPos),Bind("b",Str("B",[]),NoPos)])])
